src/persist-client/src/impl/gc.rs

```diff
@@ -130,72 +130,81 @@ impl GarbageCollector {
         // already gc'd and truncated past new_seqno_since.
 
         let path = req.shard_id.to_string();
-        let state_versions = retry_external(&metrics.retries.external.gc_scan, || async {
-            consensus.scan(&path, SeqNo::minimum()).await
+
+        // pull down the latest version of state, which contains the blobs we want to keep
+        let head = retry_external(&metrics.retries.external.gc_scan, || async {
+            consensus.head(&path).await
```

Comment:
pretty sure this has to be req.new_seqno_since :)

this could do a scan limit 1 to grab it?

Reply:
🙈 well that would have been a nasty bug to fix